src/transport/smtp/async_transport.rs

```diff
@@ -158,15 +158,48 @@ where
     /// [`AsyncSmtpTransport::starttls_relay`](#method.starttls_relay) instead,
     /// if possible.
     pub fn builder_dangerous<T: Into<String>>(server: T) -> AsyncSmtpTransportBuilder {
-        let info = SmtpInfo {
-            server: server.into(),
-            ..Default::default()
-        };
-        AsyncSmtpTransportBuilder {
-            info,
-            #[cfg(feature = "pool")]
-            pool_config: PoolConfig::default(),
-        }
+        AsyncSmtpTransportBuilder::new(server)
+    }
+
+    /// Creates a `AsyncSmtpTransportBuilder` from a connection URL
+    ///
+    /// The protocol, credentials, host and port can be provided in a single URL.
+    /// Use the scheme `smtp` for an unencrypted relay, `smtps` for SMTP over TLS
+    /// and `smtp` with the query parameter tls=required or tls=opportunistic for STARTTLS
```

Comment:
I think this part would be better explained by a table, including a warning discouraging the use `smtp` as it's unencrypted and `opportunistic` not making sense in most cases because of MITM attacks defeating it.

Reply:
Sure - I've added a table.